pkg/db/advisory.go

```diff
@@ -38,12 +39,8 @@ func (dbc Config) GetAdvisories(source, pkgName string) ([]types.Advisory, error
 		}
 
 		advisory.VulnerabilityID = vulnID
-		if v.Source != (types.DataSource{}) {
-			advisory.DataSource = &types.DataSource{
-				ID:   v.Source.ID,
-				Name: v.Source.Name,
-				URL:  v.Source.URL,
-			}
+		if !lo.IsEmpty(v.Source) {
+			advisory.DataSource = &v.Source
```

Comment:
Won't it cause any problems now? It was probably fixed in Go 1.22, but I'm still concerned about this kind of code.
https://go.dev/blog/loopvar-preview

Reply:
I managed to reproduce this case in trivy-db:
- can't upgrade to `1.21` (trivy-db requires `1.23`)
- couldn't get this case for trivy-db (simpler tests work as described in https://go.dev/blog/loopvar-preview), but it doesn't mean I didn't just miss something.

so I decided to use the old logic, because I'm 100% sure of it - https://github.com/aquasecurity/trivy-db/pull/555/commits/476cfba57f33de13e0faaa1094aa75063a839110

Reply:
I've struggled with loop variable issues many times, so even if I think it's probably already been fixed and should be fine, I still have doubts. If we can be certain that there's no issue, I think the new code is better since it's simpler. I've approved it, so I'll leave the final decision up to you.